packages/fast-components-styles-msft/src/text-action/index.ts

```diff
@@ -0,0 +1,147 @@
+    contrast,
+    Direction,
+    toPx,
+} from "@microsoft/fast-jss-utilities";
+import {
+    applyMixedColor,
+    disabledContrast,
+    ensureNormalContrast,
+    ensuresBackgroundNormal,
+    foregroundNormal,
+    hoverContrast,
+    normalContrast,
+} from "../utilities/colors";
+import { density } from "../utilities/density";
+import { defaultHeight, maxHeight, minHeight } from "../utilities/height";
+
+// Since MSFT button is already styled, we need to override in this way to alter button classes
+export const textFieldOverrides: ComponentStyles<
+    Partial<TextFieldClassNameContract>,
+    DesignSystem
+> = {
+    textField: {
+        height: "100%",
+        marginTop: "0",
+        border: "none",
+        flex: "1 0 0",
+        background: "transparent",
+        "&:hover, &:focus, &:disabled": {
+            borderColor: "transparent",
+            boxShadow: "none",
+        },
+    },
+};
+
+/**
+ * Retrieves the disabled color
+ */
+function disabledColor(config: DesignSystem): string {
+    const designSystem: DesignSystem = withDesignSystemDefaults(config);
+    return disabledContrast(
+        designSystem.contrast,
+        foregroundNormal(designSystem),
+        designSystem.backgroundColor
+    );
+}
+
+function hoverColor(config: DesignSystem): string {
+    const designSystem: DesignSystem = withDesignSystemDefaults(config);
+    return hoverContrast(
+        designSystem.contrast,
+        foregroundNormal(designSystem),
+        designSystem.backgroundColor
+    );
+}
+
+/* tslint:disable:max-line-length */
+const styles: ComponentStyles<TextActionClassNameContract, DesignSystem> = (
+    config: DesignSystem
+): ComponentStyleSheet<TextActionClassNameContract, DesignSystem> => {
+    const designSystem: DesignSystem = withDesignSystemDefaults(config);
+    const backgroundColor: string = designSystem.backgroundColor;
+    const direction: Direction = designSystem.direction;
+    const foregroundColor: string = ensureNormalContrast(
+        designSystem.contrast,
+        designSystem.foregroundColor,
+        designSystem.backgroundColor
+    );
+
+    const glyphStyles: CSSRules<{}> = {
+        height: "16px",
+        width: "auto",
+        marginTop: "8px",
+        margin: "auto",
+    };
+
+    return {
+        textAction: {
+            position: "relative",
+            height: density(defaultHeight)(designSystem),
+            minHeight: toPx(minHeight),
+            maxHeight: toPx(maxHeight),
+            margin: `${density(designSystem.designUnit * 3)(designSystem)} 0`,
+            minWidth: "92px",
+            border: `1px solid ${foregroundNormal(designSystem)}`,
+            background: ensuresBackgroundNormal,
+            borderRadius: "2px",
+            display: "flex",
+            flexDirection: "row",
+            transition: "all 0.2s ease-in-out",
+            "&:hover": {
+                borderColor: hoverColor(designSystem),
+            },
+            "&:focus-within": {
```

Comment:
The focus states of this component feel wrong because there are two tab-stops with the same focus treatment, one for the button and one for the input. I think we need to address focus differently so that I can clearly identify which element is focused

Reply:
Agreed - there is no design for this, so I am just going by the seat of my pants as it were. Any suggestions?

Reply:
I would start with not removing the focus state on the button itself.

Reply:
So after debating on the best way to solve this, I decided that instead of using `focus-within`, I would just use a focus class to control the focus UI. So when the inner `input` gets focus, an outer focus class is added to the wrapper (the `<TextAction />`. On blur, the focus class is removed. Therefore, we are not mixing focus states.